test/libsolidity/util/TestFileParserTests.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Unit tests for Solidity's test expectation parser.
+ */
+
+#include <functional>
+#include <string>
+#include <tuple>
+#include <boost/test/unit_test.hpp>
+#include <liblangutil/Exceptions.h>
+#include <test/libsolidity/SolidityExecutionFramework.h>
+
+#include <test/libsolidity/util/TestFileParser.h>
+
+using namespace std;
+using namespace dev::test;
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+vector<FunctionCall> parse(string const& _source)
+{
+	istringstream stream{_source, ios_base::out};
+	TestFileParser parser{stream};
+	return parser.parseFunctionCalls();
+}
+
+BOOST_AUTO_TEST_SUITE(TestFileParserTest)
+
+BOOST_AUTO_TEST_CASE(smoke_test)
+{
+	char const* source = R"()";
+	BOOST_CHECK_EQUAL(parse(source).size(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(simple_call_succees)
+{
+	char const* source = R"(
+		// f()
+		// -> 1
+	)";
+	auto const& calls = parse(source);
+	BOOST_CHECK_EQUAL(calls.size(), 1);
```

Comment:
Should use `BOOST_REQUIRE_EQUAL` because the test should not continue if it fails (the `.at` below will throw, but I guess it is cleaner to make the test stop properly).